### PR TITLE
JCL-361: Make it easier to modify Access Control resources

### PR DESCRIPTION
**Pull request by acoburn**
This introduces a utility function to make it easier to modify Access Control resources to support AccessGrants.

Eventually, this will become part of a stand-alone ACP module, but adding full support for ACP is beyond the scope of the 1.0 release. This, however, allows developers to interact more easily with ACRs within the 1.0 release timeframe.

A sample usage of this would be:

```java
try (final SolidRDFSource resource = client.read(uri, SolidRDFSource.class)) {
    resource.getMetadata().getAcl().ifPresent(acl -> {
        try (final SolidRDFSource acr = client.read(acl, SolidRDFSource.class)) {
            AccessGrantUtils.accessControlPolicyTriples(acl, ACL.Read, ACL.Write)
                .forEach(acr.getGraph()::add);
            client.update(acr);
        }
    });
}
```

### CODE REVIEW

File: access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantClient.java [hunk 1]
```diff
@@ -391,13 +391,13 @@ public <T extends AccessCredential> CompletionStage<List<T>> query(final URI age
         final URI type;
         final Set<String> supportedTypes;
         if (AccessGrant.class.isAssignableFrom(clazz)) {
-            type = ACCESS_GRANT;
+            type = URI.create("SolidAccessGrant");
```
Comment (NSeydoux) [inline access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantClient.java:394]: For my own understanding, in this case where the given string isn't a URI, what does the constructor do?

File: access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantUtils.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        // Access Control
+        final IRI accessControl = asIRI(URIBuilder.newBuilder(acl).fragment(UUID.randomUUID().toString()).build());
+        triples.add(rdf.createTriple(accessControl, a, asIRI(ACP.AccessControl)));
+        triples.add(rdf.createTriple(accessControl, asIRI(ACP.apply), policy));
+
+        // Access Control Resource
+        final IRI subject = asIRI(acl);
+        triples.add(rdf.createTriple(subject, a, asIRI(ACP.AccessControlResource)));
```
Comment (NSeydoux) [inline access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantUtils.java:74]: Is this triple necessary? I assumed this type would be present by default in the ACR.

**Review by NSeydoux — APPROVED** [review]

File: access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantClient.java (hunk 1 above)
Comment (acoburn) [reply access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantClient.java:394]: This is a relative URI. Perfectly valid

File: access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantUtils.java (hunk 2 above)
Comment (acoburn) [reply access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantUtils.java:74]: In the Inrupt implementation, none of the type triples are strictly necessary. That said, a different impl may require them. Furthermore, the ACR for a resource may be empty, in which case the triple will not already be present.

In short: it never hurts to be explicit, and there may be cases where it is necessary

**Pull request merged**